core/src/main/java/org/elasticsearch/script/ScriptModule.java

```diff
@@ -34,57 +35,60 @@
 import java.util.stream.Collectors;
 
 /**
- * An {@link org.elasticsearch.common.inject.Module} which manages {@link ScriptEngineService}s, as well
- * as named script
+ * Manages building {@link ScriptService} and {@link ScriptSettings} from a list of plugins.
  */
-public class ScriptModule extends AbstractModule {
-
-    protected final ScriptContextRegistry scriptContextRegistry;
-    protected final ScriptEngineRegistry scriptEngineRegistry;
-    protected final ScriptSettings scriptSettings;
-
-    public ScriptModule(ScriptEngineService... services) {
-        this(Arrays.asList(services), Collections.emptyList());
-    }
-
-    public ScriptModule(List<ScriptEngineService> scriptEngineServices,
-                        List<ScriptContext.Plugin> customScriptContexts) {
-        this.scriptContextRegistry = new ScriptContextRegistry(customScriptContexts);
-        this.scriptEngineRegistry = new ScriptEngineRegistry(scriptEngineServices);
-        this.scriptSettings = new ScriptSettings(scriptEngineRegistry, scriptContextRegistry);
-    }
+public class ScriptModule {
+    private final ScriptSettings scriptSettings;
+    private final ScriptService scriptService;
 
     /**
-     * This method is called after all modules have been processed but before we actually validate all settings. This allows the
-     * script extensions to add all their settings.
+     * Shortcut constructor for tests.
      */
-    public List<Setting<?>> getSettings() {
-        ArrayList<Setting<?>> settings = new ArrayList<>();
-        scriptSettings.getScriptTypeSettings().forEach(settings::add);
-        scriptSettings.getScriptContextSettings().forEach(settings::add);
-        scriptSettings.getScriptLanguageSettings().forEach(settings::add);
-        settings.add(scriptSettings.getDefaultScriptLanguageSetting());
-        return settings;
+    public ScriptModule(Settings settings, Environment environment, ResourceWatcherService resourceWatcherService,
```

Comment:
I find it confusing to have both a public ctor for tests, but also a static builder method. It looks like there is a only one or two uses of this ctor in tests, and most tests that need a ScriptModule should use `newTestScriptModule()`? The couple uses could just use a ctor or static method directly, instead of having a "test only" ctor?